### PR TITLE
Fixes #32295 - Puppet service disabled by default 

### PR DESCRIPTION
**Pull request by nadjaheitmann**

### CODE REVIEW

File: app/views/unattended/provisioning_templates/snippet/puppet_setup.erb [hunk 1]
```diff
@@ -152,7 +152,24 @@ export FACTER_is_installer=true
...
+echo "#!/bin/sh
+### BEGIN INIT INFO
+# Provides: puppet_start
+# Required-Start: $all
+# Required-Stop:
+# Default-Start:  2 3 5
+# Default-Stop:
+# Description: Start puppet service
+### END INIT INFO
+puppet resource service puppet enable=true" >> "${START_PUPPET_SCRIPT}"
```
Comment (ekohl) [inline app/views/unattended/provisioning_templates/snippet/puppet_setup.erb]: Do you really want this enable line inside the init script? It feels weird to do that every single time.

**Review by ekohl — COMMENTED** [review]
Could you fix the commit? Please also drop the Cherry picked part. We only apply those in stable branches and the commit mentioned can't be found in the history.

**Comment by nadjaheitmann** [comment]
[test foreman]

**Comment by nadjaheitmann** [comment]
[test katello]

File: app/views/unattended/provisioning_templates/snippet/puppet_setup.erb (hunk 1 above)
Comment (nadjaheitmann) [reply app/views/unattended/provisioning_templates/snippet/puppet_setup.erb]: Thank you for your feedback. I came up with a more elegant solution.

**Comment by nadjaheitmann** [comment]
[test upgrade]

**Review by ekohl — COMMENTED** [review]
This suggests a Puppet bug. With which version have you tried this?

**Comment by nadjaheitmann** [comment]
[test katello]

**Comment by nadjaheitmann** [comment]
We use 6.21, error occurs on Deb9/10 and Ubuntu 18/20, not on SLES12/15. I have to re-test SLES11 again, though. Seems we accidentally used an older version there.

**Comment by ekohl** [comment]
I would at least file a bug report against Puppet if it is indeed a problem.

**Comment by nadjaheitmann** [comment]
It seems to be a problem during deployment. Once the host is up, the previous command works fine (`puppet resource service puppet enable=true`).

**Comment by ekohl** [comment]
It could very well be that the installer environment isn't properly detected so it uses a wrong provider. Sometimes there's a workaround for that.

Looking at this line, it looks like it should work if it can detect the OS release:
https://github.com/puppetlabs/puppet/blob/b9240c317780df0359eb2451db0a4cb0177a353a/lib/puppet/provider/service/redhat.rb#L12

It would be interesting to look at a debug log of the command and the output of facter inside the installer.

**Comment by nadjaheitmann** [comment]
> 5272:Apr  1 21:05:31 puppet-agent[10874]: Applied catalog in 0.05 seconds
> 5273-Apr  1 21:05:42 in-target: Running in chroot, ignoring request.
> 5274:Apr  1 21:05:42 in-target: ^[[1;31mError: Execution of '/usr/sbin/update-rc.d puppet defaults' returned 1: update-rc.d: error: unable to read /etc/init.d/puppet^[[0m
> 5275:Apr  1 21:05:42 in-target: ^[[1;31mError: /Service[puppet]/enable: change from 'false' to 'true' failed: Execution of '/usr/sbin/update-rc.d puppet defaults' returned 1: update-rc.d: error: unable to read /etc/init.d/puppet^[[0m
> 5276-Apr  1 21:05:42 in-target: Running in chroot, ignoring request.
> 5277:Apr  1 21:05:42 in-target: service { 'puppet':
> 5278-Apr  1 21:05:42 in-target:   enable   => 'false',
> 5279-Apr  1 21:05:42 in-target:   provider => 'debian',
> 5280-Apr  1 21:05:42 in-target: }

Looks like this is similar to here:
https://tickets.puppetlabs.com/browse/PUP-10963

**Comment by nadjaheitmann** [comment]
I have re-tested the provisioning with SLES 11.4 and Puppet 6.21. As expected, no problem there. So it's only Ubuntu and Debian that need some workaround. 
@ekohl Does it make sense to move away from relying on puppet to find the correct service provider during installation? When SLES11.4 support runs out, `systemctl` can be used for SLES12/15 as well.

**Review by ekohl — APPROVED** [review]
Thanks!

**Comment by ezr-ondrej** [comment]
CP 2.4: #8505

**Pull request merged**